node/src/blockchain/blockchain_bridge.rs

```diff
@@ -426,223 +358,242 @@ mod tests {
     }
 
     #[test]
-    fn ask_me_about_my_transactions() {
-        let system = System::new("ask_me_about_my_transactions");
-        let block_no = 37;
-        let expected_results = vec![Transaction {
-            block_number: 42u64,
-            from: make_wallet("some_address"),
-            gwei_amount: 21,
-        }];
-        let result = Ok(expected_results.clone());
-        let wallet = make_wallet("smelly");
+    fn report_accounts_payable_returns_error_for_blockchain_error() {
         let blockchain_interface_mock = BlockchainInterfaceMock::default()
-            .retrieve_transactions_result(result)
-            .contract_address_result(TEST_DEFAULT_CHAIN.rec().contract);
-        let retrieve_transactions_parameters = blockchain_interface_mock
-            .retrieve_transactions_parameters
+            .get_transaction_count_result(Ok(web3::types::U256::from(1)))
+            .send_transaction_result(Err(BlockchainError::TransactionFailed(String::from(
+                "mock payment failure",
+            ))));
+        let transaction_count_parameters = blockchain_interface_mock
```

Comment:
Wha...?

Maybe define a local parameters list and use it the same way we're accustomed to doing?